src/app/contentful.service.ts

```diff
@@ -23,20 +24,16 @@ export class ContentfulService {
   }
 
   getPosts(query?: object): any {
-    return this.cdaClient
-      .getEntries({
+    return from(
+      this.cdaClient.getEntries({
         ...Object,
-        content_type: this.CONFIG.contentTypeIds.angularPost,
+        content_type: this.CONFIG.contentTypeIds.blogPost,
         query
       })
-      .then(posts => {
-        return of(posts.items);
-      });
+    ).pipe(map(posts => posts.items));
   }
 
   getPost(id: string): any {
-    return this.cdaClient.getEntry(id).then(post => {
-      return of(post);
-    });
+    return from(this.cdaClient.getEntry(id));
```

Comment:
Note: as mentioned this isn't lazy. To make it lazy we can use `eager()`. But that's out of scope for the article.